cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/QuerySqlTranslator.java

```diff
@@ -562,7 +562,12 @@ protected static SqlParts whereSqlForAndClause(List<Object> clause, String index
                         // This was validated during normalization.
                         predicateValue = negatedPredicate.get(operator);
                         placeholder = "?";
-                        sqlParameters.add(String.valueOf(predicateValue));
+                        if (predicateValue instanceof Boolean){
```

Comment:
I meant (for example) like this, so the API wouldn't be an issue:
```java
        List<Object> sqlParameters = new ArrayList<Object>() {
            @Override
            public boolean add(Object o) {
                         if (o instanceof Boolean){
                            Boolean bool = (Boolean) o;
                            return super.add(bool ? "1" : "0");
                        } else {
                            return super.add(String.valueOf(o));
                        }
            }
        };
```
It would avoid the need to `add(String.valueOf(predicateValue))` in so many places as well, but we'd need to take care with the `add(modulus.get(0))` since they are `Integer`.